### DISCUSSION TITLE
Build error in "Parsing java" with IntelliJ

### DISCUSSION THREAD

**Discussion by jn1z** [LearnLib/automatalib#66] [Q&A] [answered]
Hi,

I'm trying to build automatalib with the latest Community IDEA (2023.2.4), and I'm getting an error during "Parsing Java", which I haven't figured out how to work around. Any ideas? Thanks, John

Notes: 
1. "mvn compile", "mvn package", etc., work fine from the command line.
2. This is the first time I've tried building this with IDEA.
3. Steps iii) and iv) in the README don't seem to apply to the latest IDEA, but I did follow the advice here: https://stackoverflow.com/questions/60759611/import-maven-projects-automatically-checkbox-dissapear-in-intellij-eap
4. The build error has advice about "jps.track.ap.dependencies". Following this, I still get the error: https://stackoverflow.com/questions/66170975/jps-incremental-annotation-processing-is-disabled-compilation-results-on-partia

Build error (snipping out a long boilerplate message):
```
Parsing java… [automata-commons-smartcollections]
java: The class java.lang.IllegalArgumentException may be caused by the wrapped ProcessingEnvironment object.
Please pass the wrapped ProcessingEnvironment further to super.init().

   ...<snip>...

java: Workaround: to make project compile with the current annotation processor implementation, start JPS with VM option: -Djps.track.ap.dependencies=false
  When run from IDE, the option can be set in "Compiler Settings | build process VM options"
java: java.lang.IllegalArgumentException
java: java.lang.IllegalArgumentException
Errors occurred while compiling module 'automata-commons-smartcollections'
javac 17.0.9 was used to compile java sources
```

**Comment by mtf90** [comment, marked as answer]
Hi @jn1z, 

I have run into this issue as well. It is related to the switch to our [in-house annotation processor](https://github.com/LearnLib/build-tools) which uses the `DocTrees` API that is not compatible with the `ProcessingEnvironment` that IntelliJ creates. I can workaround this in the annotation processor but I still need some time to think about whether I should do this or find an alternative for the `DocTrees` API (since gradle users seem to encounter similar issues). Currently, I'm busy with the modularization of AutomataLib and LearnLib but I have it on my list as the next point after this.

For now, you should be fine with disabling the incremental annotation processing as suggested. At least this is what works for me (see below)
 
![2024-01-08-154355_1102x628_scrot](https://github.com/LearnLib/automatalib/assets/7063944/84010828-9209-4d56-980b-c171144b3996)

**Reply by jn1z** [reply]
Thank you! That worked for me. (I had tried that previously, but perhaps I had a typo.)

**Reply by mtf90** [reply]
The workaround should no longer be necessary with 2eaa2c4.
